content/docs/rfcs/43/README.md

```diff
@@ -0,0 +1,311 @@
+
+The request is made by exposing a QR code that, by default, has to be scanned by device `A`. 
+If device `A` doesn't have a camera while device `B` does, 
+[it is possible](#Rationale) to execute a slightly different pairing (with same security guarantees), 
+where `A` is exposing a QR code instead.
+
+### Employed Cryptographic Primitives
+
+- `H`: the underlying hash function, i.e. SHA-256;
+- `HKDF`: a key derivation function (based on SHA-256);
+- `Curve25519`: the underlying elliptic curve for Diffie-Hellman (DH) operations.
+
+### The `WakuPairing` Noise Handshake
+
+The devices execute a custom handshake derived from `X1X1`, 
+where they mutually exchange and authenticate their device static keys 
+by exchanging messages over the content topic
+
+```
+contentTopic = /{application-name}/{application-version}/wakunoise/1/sessions-{shard-id}/proto
+```
+
+The handshake, detailed in next section, can be summarized as:
+
+```
+WakuPairing:
+0.   <- eB              {H(sB||r), contentTopicParams, messageNametag}
+     ...
+1.   -> eA, eAeB        {H(sA||s)}   [authcode]
+2.   <- sB, eAsB        {r}
+3.   -> sA, sAeB, sAsB  {s}
+
+{}: payload,    []: user interaction
+```
+
+### Protocol Flow
+
+1. The device `B` exposes through a QR code a Base64 serialization of:
+    - An ephemeral public key `eB`;
+    - The content topic parameters `contentTopicParams = {application-name}, {application-version}, {shard-id}`.
+    - A randomly generated 8-bytes long `messageNametag`. 
+    - A commitment `H(sB||r)` for its static key `sB` where `r` is a random fixed-lenght value.
+
+2. The device `A`:
+    - scans the QR code;
+    - obtains `eB`, `contentTopicParams`, `messageNametag`, `Hash(sB|r)`;
+    - checks if `{application-name}` and `{application-version}` from `contentTopicParams` match the local application name and version: if not, aborts the pairing. 
+    - initializes the Noise handshake by passing `contentTopicParams`, `messageNametag` and `Hash(sB||r)` to the handshake prologue;
+    - executes the pre-handshake message, i.e. processes the key `eB`;
+    - executes the first handshake message over `contentTopic`, i.e. 
+        - processes and sends a Waku message containing an ephemeral key `eA`; 
+        - performs `DH(eA,eB)` (which computes a symmetric encryption key);
+        - attach as payload to the handshake message a commitment `H(sA|s)` for `A`'s static key `sA`, where `s` is a random fixed-lenght value;
+    - an 8-digits authorization code `authcode` obtained as `HKDF(h) mod 10^8` is displayed on the device, where `h`is the handshake value obtained once the first handshake message is processed.
+
+3. The device `B`:
+    - listens to messages sent to `/{application-name}/{application-version}/wakunoise/1/sessions-{shard-id}/proto` and locally filters only those with [Waku payload](https://rfc.vac.dev/spec/35/#abnf) starting with `messageNametag`. If any, continues.
```

Comment:
https://rfc.vac.dev/spec/35/#abnf specifies a specific format that does not have a `messageNameTag` at the start of the payload. I am guessing we should update 35/WAKU2-NOISE then? or should `messageNameTag` be set at the start of `transport-message`?